src/main/java/mate/academy/onlinebookstoreproject/repository/OrderItemRepository.java

```diff
@@ -0,0 +1,17 @@
+package mate.academy.onlinebookstoreproject.repository;
+
+import java.util.List;
+import mate.academy.onlinebookstoreproject.model.OrderItem;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface OrderItemRepository extends JpaRepository<OrderItem, Long> {
+    @Query("select o from OrderItem o where o.order.id = :orderId and o.order.user.id = :userId")
+    List<OrderItem> findAllByOrderId(Long orderId, Long userId);
+
+    @Query("select o from OrderItem o "
+            + "where o.order.id = :orderId and o.order.user.id = :userId and o.id = :itemId")

```

Comment:
Since Java 15 you can use [Text Blocks](https://www.baeldung.com/java-text-blocks) which looks better with this queries, but its up to you